webapp/src/components/tables/existingConfigTable.tsx

```diff
@@ -0,0 +1,187 @@
+import React, {useState} from 'react';
+
+import {FormGroup, Table, ButtonGroup, Button, OverlayTrigger, Tooltip} from 'react-bootstrap';
+
+import ActionModal from '../modals/actionModal';
+import DeleteModal from '../modals/deleteModal';
+import ConfigModal from '../modals/configModal';
+
+import {DeleteSvg, EditSvg, ViewSvg} from '../svgIcons/svg';
+
+import './styles.scss';
+
+type Props = {
+    onChange: (config: Configs[]) => void;
+    value: Configs[];
+}
+
+const ExistingConfigTable = ({value, onChange}: Props) => {
+    const [isViewVisible, setIsViewVisible] = useState(false);
+    const [isDeleteVisible, setIsDeleteVisible] = useState(false);
+    const [isEditVisible, setIsEditVisible] = useState(false);
+    const [isAddVisible, setIsAddVisible] = useState(false);
+
+    const [configIndex, setConfigIndex] = useState(0);
+
+    const handleView = (index: number) => {
+        setConfigIndex(index);
+        setIsViewVisible(true);
+    };
+
+    const handleDelete = (index: number) => {
+        setConfigIndex(index);
+        setIsDeleteVisible(true);
+    };
+
+    const handleEdit = (index: number) => {
+        setConfigIndex(index);
+        setIsEditVisible(true);
+    };
+
+    const handleAdd = () => setIsAddVisible(true);
+
+    return (
+        <div className='config'>
+            {
+                isViewVisible &&
+                    <ActionModal
+                        visibility={isViewVisible}
+                        setVisibility={setIsViewVisible}
+                        config={value}
+                        configIndex={configIndex}
+                    />
+            }
+            {isDeleteVisible &&
+                <DeleteModal
+                    visibility={isDeleteVisible}
+                    setVisibility={setIsDeleteVisible}
+                    config={value}
+                    configIndex={configIndex}
+                    onChange={onChange}
+                />
+            }
+            {isEditVisible &&
+                <ConfigModal
+                    visibility={isEditVisible}
+                    setVisibility={setIsEditVisible}
+                    configIndex={configIndex}
+                    config={value}
+                    onChange={onChange}
+                    modalHeader='Edit Config'
+                />
+            }
+            {isAddVisible &&
+                <ConfigModal
+                    visibility={isAddVisible}
+                    setVisibility={setIsAddVisible}
+                    configIndex={null}
+                    config={value}
+                    onChange={onChange}
+                    modalHeader='Add Config'
+                />
+            }
+            <FormGroup>
```

Comment:
```suggestion

            <FormGroup>
```
use a new line to separate a block/section